lib/registration.pm

```diff
@@ -190,7 +190,12 @@ sub fill_in_registration_data {
     if (sle_version_at_least('15') && check_var('DISTRI', 'sle')) {
         my $modules_needle = "modules-preselected-" . get_required_var('SLE_PRODUCT');
         if (get_var('UPGRADE') || get_var('PATCH')) {
-            check_screen($modules_needle, 5);
+            if (is_system_upgrading) {
+                unless (check_screen($modules_needle, 5)) {
+                    record_soft_failure('bsc#1070031: Module Selection is missing in upgrade');
+                    return;
+                }
+            }
```

Comment:
How would you like instead

```perl
return record_soft_failure('bsc#1070031: Module Selection is missing in upgrade') if is_system_upgrading && !check_screen($modules_needle, 5);
```

or at least combine the if+unless?

Reply:
Yes, less condition levels and more concise

Reply:
Updated according to your comment and verified with new test run: http://openqa-apac1.suse.de/tests/218
Thanks.

Reply:
@QingmingSu have to resolve conflicts becase of #3959 merged